# Understanding Air Quality in Canada and Its Impact on Environment
Jenny Lee, September 2022

```{note}
Kindly note that this web page solely offers a synopsis of the **Air Quality in Canada** project. The comprehensive set of scripts and codes used in the project is available in the corresponding [GitHub repository](https://github.com/jlee2843/data-visualizations-portfolio/tree/main/air-quality).
```

## Project Outline

In [27]:
import pandas as pd 
import numpy as np
import IPython
from IPython.display import YouTubeVideo
from IPython.display import HTML
from myst_nb import glue
from schemdraw.flow import *
from schemdraw import flow
import plotly.io as pio
import plotly
import plotly.express as px
import plotly.graph_objects as go
import plotly.figure_factory as ff
import matplotlib.pyplot as plt
from plotly.subplots import make_subplots
import plotly.offline as py
from IPython import display
import scipy.stats as stats

```{admonition} Data science toolbox used in this project
:class: tip, dropdown
- Data analysis
- Data cleaning and wrangling
- Interactive visualizations with `Plotly`
```

In this project, I integrated three different types of datasets: air pollutant emissions, wildfires in Canada, and Canadian population estimates. The original data sources were as follows:

- **Air Pollution Emissions Across Provinces**: Canada's Air Pollutant Emissions Inventory[^ref1]
- **Wildfires in Canada**: Canadian National Fire Database[^ref2]
- **Canadian Population Estimates**: Population Estimates from Statistics Canada[^ref3]

The Government of Canada[^ref4]  had identified sulfur oxides, nitrogen oxides, volatile organic compounds, particulate matter, carbon monoxide, ammonia, and ground-level ozone as the most common air contaminants in Canada.

Canada's Air Pollutant Inventory[^ref5] listed a total of thirteen air pollutants. For this project, the focus was on sulfur oxides, nitrogen oxides, volatile organic compounds, carbon monoxide, and ammonia—the pollutants recognized as the most common in Canada.

## Overall Process

### Gathering Provincial Pollutant Emissions per Capita

To investigate the air quality in Canada from the past to the present, we start by identifying pollutants that significantly impact air quality and health. We have chosen $\text{NO}_X$, $\text{NH}_3$, $\text{CO}$, $\text{SO}_X$, and $\text{VOC}$ as major pollutants for closer examination. 

First, we start by gathering data from Statistics Canada's Air Pollutant Emissions Inventory [^ref1] across all provinces in Canada. Data is collected from 1990 to 2020. 

In [5]:
pd.read_csv("projects/environmental-health/csv/provincial_df.csv")

Province Pollutant          1990          1991          1992          1993  \
0        AB       NH3  9.526164e+04  9.639261e+04  1.008097e+05  1.028888e+05   
1        AB        CO  1.810389e+06  1.674240e+06  1.694347e+06  1.670384e+06   
2        AB       SOX  5.124052e+05  5.241442e+05  5.647178e+05  5.714739e+05   
3        AB       NOX  6.132843e+05  5.857053e+05  6.094636e+05  6.408844e+05   
4        AB       VOC  6.437522e+05  6.246338e+05  6.424776e+05  6.543844e+05   
5        BC       NH3  2.299599e+04  2.231089e+04  2.288312e+04  2.342389e+04   
6        BC        CO  2.340344e+06  2.353114e+06  2.155700e+06  2.363510e+06   
7        BC       SOX  1.107474e+05  9.068985e+04  7.935032e+04  7.742206e+04   
8        BC       NOX  2.859677e+05  2.708281e+05  2.621269e+05  2.718991e+05   
9        BC       VOC  3.749732e+05  3.767629e+05  3.551831e+05  3.902692e+05   
10       ON       NH3  1.105196e+05  1.068000e+05  1.054187e+05  1.028781e+05   
11       ON        CO  3.269271e+06  3.288746e+06  3.289638e+06  3.326173e+06   
12       ON       SOX  1.123223e+06  1.040497e+06  8.500874e+05  7.479657e+05   
13       ON       NOX  6.026165e+05  5.921593e+05  6.060585e+05  5.953700e+05   
14       ON       VOC  8.527794e+05  8.370145e+05  8.394441e+05  8.546847e+05   
15       MB       NH3  3.804731e+04  4.018156e+04  4.177450e+04  4.321955e+04   
16       MB        CO  3.858356e+05  3.818272e+05  3.783853e+05  3.678765e+05   
17       MB       SOX  5.091390e+05  4.031601e+05  4.182844e+05  4.003879e+05   
18       MB       NOX  7.257550e+04  6.755345e+04  6.683440e+04  6.732315e+04   
19       MB       VOC  9.606369e+04  9.491176e+04  9.446720e+04  9.268034e+04   
20       NB       NH3  4.594717e+03  4.562918e+03  4.680083e+03  4.605949e+03   
21       NB        CO  3.161127e+05  3.074096e+05  3.128255e+05  3.125358e+05   
22       NB       SOX  1.093135e+05  1.100564e+05  1.142301e+05  1.138806e+05   
23       NB       NOX  7.872505e+04  7.444946e+04  7.275684e+04  6.994048e+04   
24       NB       VOC  8.213297e+04  8.067511e+04  8.186091e+04  8.228939e+04   
25       NS       NH3  4.863271e+03  4.869619e+03  4.881470e+03  4.836537e+03   
26       NS        CO  3.084968e+05  2.900437e+05  2.958940e+05  2.984767e+05   
27       NS       SOX  2.031522e+05  2.012478e+05  2.012421e+05  1.973347e+05   
28       NS       NOX  8.288278e+04  8.065137e+04  8.275911e+04  8.377182e+04   
29       NS       VOC  7.023577e+04  6.794989e+04  6.976300e+04  7.036716e+04   
30       YT       NH3  1.443085e+01  1.672027e+01  1.750614e+01  1.775272e+01   
31       YT        CO  1.638891e+04  1.661362e+04  1.704833e+04  1.676406e+04   
32       YT       SOX  6.391755e+02  8.939871e+02  1.000441e+03  1.343235e+03   
33       YT       NOX  3.749376e+03  3.961127e+03  4.384269e+03  3.192706e+03   
34       YT       VOC  2.366576e+03  2.463797e+03  2.671464e+03  2.571805e+03   
35       QC       NH3  6.495599e+04  6.351689e+04  6.329931e+04  6.361111e+04   
36       QC        CO  2.255181e+06  2.215428e+06  2.298285e+06  2.312405e+06   
37       QC       SOX  2.468858e+05  2.212172e+05  2.213172e+05  2.265515e+05   
38       QC       NOX  3.116497e+05  2.896130e+05  2.972382e+05  3.061471e+05   
39       QC       VOC  4.928824e+05  4.723928e+05  4.845894e+05  4.949882e+05   
40       SK       NH3  4.859078e+04  4.770903e+04  5.450196e+04  5.639792e+04   
41       SK        CO  5.226343e+05  5.240805e+05  5.217318e+05  5.250493e+05   
42       SK       SOX  1.029411e+05  1.088700e+05  1.136843e+05  1.219465e+05   
43       SK       NOX  1.492980e+05  1.563658e+05  1.606234e+05  1.705858e+05   
44       SK       VOC  1.868643e+05  1.880496e+05  1.930209e+05  2.031881e+05   
45       NL       NH3  1.029078e+03  1.013347e+03  1.025213e+03  1.014075e+03   
46       NL        CO  1.854559e+05  1.759779e+05  1.754958e+05  1.809863e+05   
47       NL       SOX  6.882082e+04  6.989177e+04  8.483419e+04  1.020158e+05   
48       NL       NOX  5.204185e+0

Next, we examine provincial emissions of air pollutants per capita. To achieve this, we first need to merge the provincial air pollutants data and population per capita data[^ref3].

We conduct additional data cleaning on the two datasets by renaming some columns. The resulting dataset encompasses provincial air pollutant emissions and population estimates for each year from 1990 to 2020. Subsequently, we divide air pollutant emissions' `Value` by population's `Per Capita` to obtain emission `Per Capita`.

Then, we take a look into the mean and air pollutants across Canadian provinces per capita. 

In [13]:
pd.read_csv("projects/environmental-health/csv/pollutant_capita_mean.csv")

Province Pollutant      Mean
0        AB        CO  0.426834
1        AB       NH3  0.038578
2        AB       NOX  0.209131
3        AB       SOX  0.131820
4        AB       VOC  0.179616
5        BC        CO  0.340157
6        BC       NH3  0.005356
7        BC       NOX  0.064060
8        BC       SOX  0.018102
9        BC       VOC  0.063772
10       MB        CO  0.226906
11       MB       NH3  0.043935
12       MB       NOX  0.050566
13       MB       SOX  0.245936
14       MB       VOC  0.066056
15       NB        CO  0.301417
16       NB       NH3  0.006092
17       NB       NOX  0.077988
18       NB       SOX  0.099255
19       NB       VOC  0.074919
20       NL        CO  0.225439
21       NL       NH3  0.001857
22       NL       NOX  0.092253
23       NL       SOX  0.086396
24       NL       VOC  0.065563
25       NS        CO  0.217794
26       NS       NH3  0.004562
27       NS       NOX  0.074748
28       NS       SOX  0.147623
29       NS       VOC  0.053418
30       ON        CO  0.189770
31       ON       NH3  0.008145
32       ON       NOX  0.041102
33       ON       SOX  0.039897
34       ON       VOC  0.049476
35       PE        CO  0.264289
36       PE       NH3  0.022612
37       PE       NOX  0.044107
38       PE       SOX  0.014450
39       PE       VOC  0.056954
40       QC        CO  0.224545
41       QC       NH3  0.008525
42       QC       NOX  0.034874
43       QC       SOX  0.027408
44       QC       VOC  0.047672
45       SK        CO  0.414566
46       SK       NH3  0.075559
47       SK       NOX  0.158855
48       SK       SOX  0.124684
49       SK       VOC  0.229759

### Enviromental Impact Caused By Air Pollution

To investigate the environmental implications of air pollution, we examine the frequency of wildfires in Canada and study how their occurrences are related to air pollution.

The Canadian National Fire Database[^ref2] provides diverse information on wildfires in Canada. We will explore the total number of wildfires and the total area burned by province. Using this information, we can assess whether the `Number of Fires` or `Area Burned (hectares)` correlates better with PM2.5 emissions.

Below are the resulting dataframes after undergoing some data cleaning.

In [21]:
pm25_fire = pd.read_csv("projects/environmental-health/csv/pm25_fire.csv").drop("Unnamed: 0", axis=1)
pm25_fire

Province  Total Emission  Number of Fire  Area Burned (hectre)
0       AB    1.330822e+07         38015.0          5.923754e+06
1       BC    2.600960e+06         56698.0          5.123188e+06
2       MB    3.077027e+06         10953.0          6.672828e+06
3       NB    6.211207e+05          9985.0          2.908940e+04
4       NL    4.874982e+05          2937.0          4.148917e+05
5       NS    6.455169e+05          8841.0          2.159480e+04
6       ON    5.625880e+06         35041.0          4.865412e+06
7       PE    1.235335e+05            11.0          2.160128e+01
8       QC    3.890464e+06         21222.0          8.173951e+06
9       SK    1.422373e+07         16964.0          1.488577e+07

## Final Deliverable

First, we will examine a **stacked bar graph** illustrating emissions per capita across all provinces. This graph also enables us to observe annual variations in pollutant emissions. We have incorporated an interactive feature using the `Plotly` library. To explore a specific year, simply drag the button on the slider bar located at the bottom of the graph.

In [17]:
with open('projects/environmental-health/figure/stacked_bar.json', 'r') as f:
    fig = pio.from_json(f.read())
fig.show()

Second, we construct a **bar graph** to illustrate the cumulative contribution of air pollutants for each province.

In [18]:
with open('projects/environmental-health/figure/bar_fig.json', 'r') as f:
    fig = pio.from_json(f.read())
fig.show()

Lastly, we create a **pie graph** to visualize provincial contributions to total air pollutant emissions. This pie graph facilitates a simpler observation of the provincial breakdown of pollutant emissions.

In [19]:
with open('projects/environmental-health/figure/pie_fig.json', 'r') as f:
    fig = pio.from_json(f.read())
fig.show()

Now we examine the combined wildfire and PM2.5 data. Here, we've created subplots of scatterplots to illustrate any relationships present between the two datasets. We observe similar trends in the wildfire data and PM2.5 emission data, as expected. However, what do these patterns truly signify?

In [20]:
with open('projects/environmental-health/figure/fire_fig.json', 'r') as f:
    fig = pio.from_json(f.read())
fig.show()

In [25]:
# calculating correlational coefficient for the occurance of wildfire and PM2.5 emission.
corr, pval=stats.pearsonr(pm25_fire["Total Emission"],pm25_fire["Area Burned (hectre)"])
print("Correlation coefficient for total area burnt and total PM2.5 emission: " + str(corr))
corr, pval=stats.pearsonr(pm25_fire["Total Emission"],pm25_fire["Number of Fire"])
print("Correlation coefficient for number of fire and total PM2.5 emission: " + str(corr))

Correlation coefficient for total area burnt and total PM2.5 emission: 0.7947497297236447
Correlation coefficient for number of fire and total PM2.5 emission: 0.38717314196102626


The code above has confirmed that the total number of wildfires and PM2.5 emissions exhibit a correlation value of $0.795$, whereas the total area burnt by wildfires and PM2.5 emissions show a correlation value of $0.387$. Therefore, it can be concluded that the total area burnt by wildfires is a better predictor for PM2.5 emissions.

## References
[^ref1]: Environment and Climate Change Canada. (n.d.). Canada's Air Pollutant Emissions Inventory. Retrieved September 2021, from https://data.ec.gc.ca/data/substances/monitor/canada-s-air-pollutant-emissions-inventory/APEI_Tables_Canada_Provinces_Territories/?lang
[^ref2]: Canadian Forest Service. (n.d.). Canadian National Fire Database. Retrieved September 2021, from https://cwfis.cfs.nrcan.gc.ca/ha/nfdb
[^ref3]: Statistics Canada. (n.d.). Population Estimates. Retrieved September 2021, from https://www150.statcan.gc.ca/t1/tbl1/en/tv.action?pid=1710000901
[^ref4]: Government of Canada. (n.d.). Common Contaminants - Air Pollution. Retrieved September 2021, from https://www.canada.ca/en/environment-climate-change/services/air-pollution/pollutants/common-contaminants.html
[^ref5]: Environment and Climate Change Canada. (n.d.). Canada's Air Pollutant Inventory. Retrieved September 2021, from https://data.ec.gc.ca/data/substances/monitor/canada-s-air-pollutant-emissions-inventory/APEI_Tables_Canada_Provinces_Territories/?lang